backend/src/main/java/fk/prof/backend/http/BackendHttpVerticle.java

```diff
@@ -71,20 +83,26 @@ public void start(Future<Void> fut) {
 
     Router router = setupRouting();
     workIdsInPipeline = vertx.sharedData().getLocalMap("WORK_ID_PIPELINE");
-    vertx.createHttpServer(HttpHelper.getHttpServerOptions(configManager.getBackendHttpServerConfig()))
+    server = vertx.createHttpServer(HttpHelper.getHttpServerOptions(configManager.getBackendHttpServerConfig()))
         .requestHandler(router::accept)
         .listen(configManager.getBackendHttpPort(), http -> completeStartup(http, fut));
   }
 
+  @Override
+  public void stop(Future<Void> stopFuture) throws Exception {
+    logger.info("BackendHttpVerticle stopping. Stopping backend server");
```

Comment:
@gauravAshok automatic cleanup of http server/clients/timers happens when verticle is undeployed. Refer vertx docs: http://vertx.io/docs/vertx-core/java/
This and similar changes in leaderhttpverticle should not be required. Unless we are noticing some bug in vertx which is not closing the server